x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/ThreadSafeStopWatch.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.action;
+
+import org.elasticsearch.core.TimeValue;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple, thread-safe stop watch for timing a single action.
+ * Allows to stop the time for building a response and to log it at a later point.
+ */
+class ThreadSafeStopWatch {
```

Comment:
No need for this class - it's used in only one place where the thread-safety is not needed.

Reply:
This PR only moves this class to a new place. It was added based on feedback that we should log the same time that the user also sees, which requires a state change (stopping the time when the query response was built), which raised (minor) thread safety concerns (I guess `onResponse` and `onFailure` could be triggered concurrently, but both need to access the stopwatch).

If it's okay, I'd like not to drop this class in this PR as it would affect how we measure execution time of successful queries.